src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/useLifecycle.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+import { useEffect, useState, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { batchActions } from 'redux-batched-actions';
+import { getOpenDataEntryActions } from '../DataEntry';
+import { type TrackerProgram, type RenderFoundation } from '../../../metaData';
+import type { OrgUnit } from '../common.types';
+
+export const useLifecycle = (
+    program: TrackerProgram,
+    formFoundation: RenderFoundation,
+    orgUnit?: OrgUnit,
+    dataEntryId: string,
+    itemId: string,
+) => {
+    const [ready, setReadyState] = useState(false);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (orgUnit) {
+            dispatch(batchActions([
+                ...getOpenDataEntryActions(program, formFoundation, orgUnit, dataEntryId, itemId),
+            ]));
+        }
+    }, [dispatch, program, formFoundation, orgUnit, dataEntryId, itemId]);
+
+    // temporary logic until we clean up the data from the redux store
```

Comment:
Hey @JoakimSM, 
Can you create a Jira ticket so we don't forget to clean this temporary logic? 
Thanks!

Reply:
Thanks for the reminder. Ticket created.